zebra-state/src/error.rs

```diff
@@ -49,6 +49,10 @@ pub enum QueueAndCommitError {
     #[non_exhaustive]
     Duplicate { block_hash: block::Hash },
 
+    #[error("already finished committing checkpoint verified blocks: dropped duplicate block, block is already committed to the state")]
+    #[non_exhaustive]
+    AlreadyCommitted,
```

Comment:
Optional: I think it would be nice to merge the `AlreadyCommitted`, `Replaced` and `ReplacedByNewer` variants into `Duplicate` and add a `source` or `location` field to `Duplicate` specifying if the duplicate was found in the queued blocks, sent blocks (`SentHashes`), non-finalized state, or finalized state. Feel free to resolve this comment if you want to leave that for a follow-up PR.